packages/runtime-vapor/src/dom/templateRef.ts

```diff
@@ -18,25 +22,30 @@ import { warn } from '../warning'
 import { queuePostFlushCb } from '../scheduler'
 
 export type NodeRef = string | Ref | ((ref: Element) => void)
+export type RefEl = Element | ComponentInternalInstance
 
 /**
  * Function for handling a template ref
  */
-export function setRef(el: Element, ref: NodeRef, refFor = false) {
+export function setRef(el: RefEl, ref: NodeRef, refFor = false) {
   if (!currentInstance) return
   const { setupState, isUnmounted } = currentInstance
 
   if (isUnmounted) {
     return
   }
 
+  const refValue = isVaporComponent(el) ? el.exposed || currentInstance : el
```

Comment:
Why here fallback to `currentInstance`?

Reply:
Since `core` fallback to `publicInstance`, I think we need to fallback something too, so I just fallback to `currentInstance` here

Reply:
should be child instance here